### ISSUE TITLE
feat(point_processes): log-space Gauss–Hermite option for the mark integral in retention_compensator

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/xtremax#103]
## Problem / Request
`retention_compensator` (`src/xtremax/point_processes/primitives/thinning.py:160`) evaluates the thinned-rate term `∫ (1 − p(t, m)) λ(t) dt` by drawing `n_mark_samples = 8` Monte-Carlo marks from the base mark distribution at each quadrature node (`:236`). For heavy-tailed marks (lognormal / Pareto sizes) combined with a saturating retention (a detection-probability curve `p(m) → 1` for large `m`), 8 samples is a noisy estimate of `E_m[1 − p(m)]`, and the resulting `ThinningProcess.log_prob` is a pseudo-marginal, key-dependent target — usable with NUTS for a fixed key, but with extra variance in the gradient that scales with the tail weight.

A consumer (jejjohnson/plumax Tier V, a temporally-marked point process of methane emission events thinned by a satellite detection probability) needs a deterministic, low-variance path: when the mark distribution is lognormal in `log m`, `E_m[g(m)]` is a Gauss–Hermite quadrature in log-space with ≤ 16 nodes, which is exact for polynomial `g` and avoids the underflow that linear-space quadrature hits when the tail index exceeds ~2.

## User Story
> As a user fitting a thinned marked process with lognormal marks, I want `retention_compensator(..., mark_integral="gauss_hermite")` so the compensator is deterministic and the NUTS target is a true (not pseudo-marginal) likelihood.

## Proposed API
```python
def retention_compensator(
    retention_fn, conditional_intensity_fn, history, T, n_points=100,
    mark_sampler_fn=None, mark_sample_key=None, n_mark_samples=8,
    *, mark_integral: Literal["mc", "gauss_hermite"] = "mc",
    mark_log_moments: Callable[[t, history], tuple[Array, Array]] | None = None,  # (μ_log, σ_log) at t; required for "gauss_hermite"
    n_quadrature_nodes: int = 16,
) -> Array: ...
```
Internally for `"gauss_hermite"`: nodes/weights from `numpy.polynomial.hermite_e.hermegauss(n)`, `m_k = exp(μ + σ x_k)`, `E[1 − p] ≈ Σ_k w_k (1 − p(t, m_k)) / √(2π)`; everything else in the function unchanged. `ThinningProcess` forwards the new keyword.

## Implementation Steps
- [ ] Add the `mark_integral` branch in `retention_compensator`; keep `"mc"` the default so existing behaviour and tests are unchanged
- [ ] Forward the keyword through `ThinningProcess.__init__` / `.log_prob` (`operators/thinning.py`) and the NumPyro wrapper
- [ ] Docstring note on when GH is exact / preferred (lognormal marks, saturating retention) and its limitation (needs log-moments; non-lognormal marks stay on MC)

## Definition of Done
- [ ] For lognormal marks and a logistic retention, GH with 16 nodes matches MC with 10⁵ samples to < 0.5 % and is bit-reproducible across keys
- [ ] `jax.grad` of `ThinningProcess.log_prob` w.r.t. retention parameters is finite under GH for a Pareto-like tail (σ_log = 2) where MC at 8 samples has visibly higher variance
- [ ] `make test-fast && make lint && make typecheck` green

## Testing
- [ ] `tests/point_processes/test_primitives_thinning.py::test_gauss_hermite_matches_mc_lognormal`
- [ ] `tests/point_processes/test_primitives_thinning.py::test_gauss_hermite_deterministic`

## Relationships
- Related: jejjohnson/plumax#108 (importance-corrected TMTPP likelihood — the downstream consumer), jejjohnson/plumax#106
